### PR TITLE
Implement the api doc for create device-groups

### PR DESCRIPTION
**Pull request by acosferreira**
# Description
In the context of the swaggo lib implement the api doc for thirdpartyrepo endpoints.
FIXES:
THEEDGE-3379
THEEDGE-3386

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Documentation update
- [ ] Tests update
- [ ] Refactor

### CODE REVIEW

**Review by de1987 — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -58,3 +58,6 @@ go/pkg/mod/*
 /build/
 /coverage.json
 /coverage.html
+
+#docfiles
+api/
```
Comment (mgold1234) [inline .gitignore:63]: please remove this file from pr

File: pkg/routes/devicegroups.go [hunk 2]
```diff
@@ -304,15 +304,16 @@ func GetAllDeviceGroups(w http.ResponseWriter, r *http.Request) {
...
+// @ID 			 CreateDeviceGroup
+// @Description  Create Device Group for an account.
 // @Tags         Device Groups
 // @Accept       json
 // @Produce      json
-// @Param        body	body	models.Image	true	"request body"
-// @Success      200 {object} models.SuccessPlaceholderResponse
-// @Failure      400 {object} errors.BadRequest
-// @Failure      500 {object} errors.InternalServerError
+// @Param        body	body models.CreateDeviceGroupAPI	true	"request body"
```
Comment (mgold1234) [inline pkg/routes/devicegroups.go:313]: those params are missing:
       parameters:
         - name: sort_by
           in: query
           description: "fields: created_at, name, updated_at. To sort DESC use - before the fields."
           schema:
             type: string
         - name: name
           in: query
           description: "field: filter by name"
           schema:
             type: string
         - name: created_at
           in: query
           description: "field: filter by create date"
           schema:
             type: string
         - name: updated_at
           in: query
           description: "field: filter by update date"
           schema:
             type: string
         - name: limit
           in: query
           description: 'field: return number of groups until limit is reached.'
           schema:
             type: integer
         - name: offset
           in: query
           description: 'field: return number of groups beginning at the offset.'
           schema:
             type: integer

**Review by mgold1234 — COMMENTED** [review]
looks good

**Comment by mgold1234** [comment]
I think its better to squash all commits to one commit

File: pkg/routes/devicegroups.go (hunk 2 above)
Comment (acosferreira) [reply pkg/routes/devicegroups.go:313]: @mgold1234 this is the endpoint to post
I dont use this parameters for the create, i believe should be added on the list of groups.
What do you think?

File: .gitignore (hunk 1 above)
Comment (acosferreira) [reply .gitignore:63]: I've added once we should not submit the changed files after run the make swaggo. 
This will prevent us to submit wrong

Comment (de1987) [reply .gitignore:63]: @mgold1234 As the `/api` directory should not be committed, I think it makes sense to be here.

Comment (loadtheaccumulator) [reply .gitignore:63]: I'm thinking the .gitignore will work great. We can override it if we need to update the README file.

File: pkg/routes/devicegroups.go (hunk 2 above)
Comment (loadtheaccumulator) [reply pkg/routes/devicegroups.go:313]: @mgold1234 it looks like the query in your comment might have been copied from the GET section above this POST. While it's possible to use it in a POST, the query would typically be associated with a GET.

https://github.com/RedHatInsights/edge-api/blob/a2fa0f7fe2e078050f5f05dbd2d8eb474f9e5a18/cmd/spec/path.yaml#L1611-L1638

File: pkg/models/devicegroups_api.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+// DeviceForDeviceGroupAPI is a device array expected to create groups with devices
+type DeviceForDeviceGroupAPI struct {
+	UUID string `json:"UUID" example:"68485bb8-6427-40ad-8711-93b6a5b4deac"`
+} // Device
+
+// DeviceGroupDevicesAPI is the /device-group return endpoint struct for openapi.json auto-gen
+type DeviceGroupDevicesAPI struct {
+	Account     string `json:"Account"`
+	OrgID       string `json:"org_id"`
```
Comment (loadtheaccumulator) [inline pkg/models/devicegroups_api.go:28]: Is Account or OrgID necessary for a consumer of the endpoint?
It's typically provided via header.

**Review by loadtheaccumulator — CHANGES_REQUESTED** [review]
I just have the question about the Account and OrgID.

File: pkg/models/devicegroups_api.go (hunk 3 above)
Comment (acosferreira) [reply pkg/models/devicegroups_api.go:28]: the account and orgId is being exposed in the return of post.
This struct i for the 200 response, should we remove that?

Comment (loadtheaccumulator) [reply pkg/models/devicegroups_api.go:28]: awesome. thanks.

**Review by loadtheaccumulator — APPROVED** [review]

**Review by loadtheaccumulator — CHANGES_REQUESTED** [review]
Please squash the commits.

**Review by loadtheaccumulator — APPROVED** [review]
lgtm

**Review by ldjebran — APPROVED** [review]
/lgtm

**Pull request merged**